modules/calib3d/src/solvepnp.cpp

```diff
@@ -311,9 +311,34 @@ bool solvePnPRansac(InputArray _opoints, InputArray _ipoints,
 
     opoints_inliers.resize(npoints1);
     ipoints_inliers.resize(npoints1);
-    result = solvePnP(opoints_inliers, ipoints_inliers, cameraMatrix,
+
+    try
+    {
+        result = solvePnP(opoints_inliers, ipoints_inliers, cameraMatrix,
                       distCoeffs, rvec, tvec, useExtrinsicGuess,
                       (flags == SOLVEPNP_P3P || flags == SOLVEPNP_AP3P) ? SOLVEPNP_EPNP : flags) ? 1 : -1;
+    }
+    catch(const cv::Exception& e)
+    {
+        CV_LOG_DEBUG(e.what());
+        CV_LOG_DEBUG(NULL, "Broken implementation for DLT algorithm needs at least 6 points for pose estimation. Fallback to EPnP.");
```

Comment:
Log message here is inaccurate. You cannot just presume that the Exception is caused by DLT algorithm.